### PR TITLE
feat: added UI for homepage

### PR DESCRIPTION
**Pull request by muskanbagrecha**

### CODE REVIEW

File: src/pages/homepage/components/who-to-follow/WhoToFollow.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    avatar: "https://i.pravatar.cc/300?img=8",
+  });
+  return (
+    <div className="w-1/5 dark:bg-gray-800 rounded-lg mx-2 hidden md:flex lg:flex">
+      <div className="overflow-y-auto p-3 md:py-4 lg:py-4 bg-gray-50 dark:bg-gray-800 rounded-lg sticky top-20 w-full">
+        <p className="dark:text-white transition duration-75 text-base font-normal text-center">
+          Who to follow?
+        </p>
+        <div>
+          {data.map((user) => (
```
Comment (zoef-yt) [inline src/pages/homepage/components/who-to-follow/WhoToFollow.tsx:14]: Destructure user

File: src/pages/homepage/components/post/Post.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+.post{
+    min-height: 20rem;
+}
+
+.post header{
+    display: flex;
+    align-items: center;
+    /* padding: 0 8px; */
```
Comment (zoef-yt) [inline src/pages/homepage/components/post/Post.css:12]: remove the comments

File: src/pages/homepage/components/feed-navigation/FeedNavigation.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+              >
+                <path
+                  fill-rule="evenodd"
+                  d="M3 3a1 1 0 00-1 1v12a1 1 0 102 0V4a1 1 0 00-1-1zm10.293 9.293a1 1 0 001.414 1.414l3-3a1 1 0 000-1.414l-3-3a1 1 0 10-1.414 1.414L14.586 9H7a1 1 0 100 2h7.586l-1.293 1.293z"
+                  clip-rule="evenodd"
+                ></path>
+              </svg>
+              <span className="flex-1 ml-3 whitespace-nowrap">Sign In</span>
+            </a>
+          </li> */}
```
Comment (zoef-yt) [inline src/pages/homepage/components/feed-navigation/FeedNavigation.tsx:97]: remove commented code.

File: src/pages/homepage/components/feed-navigation/FeedNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            <Link
+              to="/"
+              className="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"
+            >
+              <Home className="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" />
+              <span className="nav-label flex-1 ml-3 whitespace-nowrap">
+                Feed
+              </span>
+            </Link>
+          </li>
```
Comment (zoef-yt) [inline src/pages/homepage/components/feed-navigation/FeedNavigation.tsx:27]: You can convert these into one component as you are here.

File: src/context/theme-context/ThemeContext.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import { createContext, useContext, useEffect, useState } from "react";
```
Comment (zoef-yt) [inline src/context/theme-context/ThemeContext.tsx:1]: Question: I can see you have theme context but the switch of the theme is not happening bug or not applied yet?

File: src/index.css [hunk 6]
```diff
@@ -76,4 +79,27 @@ html {
     margin: 0;
     padding: 0;
     background-color: var(--color-current);
+}
+
+nav.navigation {
+    background-color: #D2CBE7;
```
Comment (zoef-yt) [inline src/index.css:85]: You var for colour.

File: src/index.tsx [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
+  document.getElementById("root") as HTMLElement
 );
 root.render(
   <React.StrictMode>
-    <App />
+    <BrowserRouter>
+      <AppPovider>
+        <App />
+      </AppPovider>
+    </BrowserRouter>
```
Comment (zoef-yt) [inline src/index.tsx:17]: Appreciation: Nice abstraction.

File: src/pages/homepage/components/create-post/CreatePost.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+      </div>
+      <div className="w-full">
+        <textarea
+          placeholder="What's on your mind?"
+          className="rounded  resize-y w-full p-2 bg-primary"
+        />
+        <footer className="flex items-center justify-center gap-12 child-hover:cursor-pointer child-hover:text-primary-700">
+          <span>
+            <Gallery />
+          </span>
```
Comment (zoef-yt) [inline src/pages/homepage/components/create-post/CreatePost.tsx:16]: These extra span can be removed.

File: src/pages/homepage/components/feed-navigation/FeedNavigation.css [hunk 9]
```diff
@@ -0,0 +1,58 @@
+.feed-navigation{
+    justify-content: space-between;
+    position: relative;
+    position: sticky;
+    bottom: 8px;
+    z-index: 2;
```
Comment (zoef-yt) [inline src/pages/homepage/components/feed-navigation/FeedNavigation.css:6]: Would prefer if you use var for z-index just an opinion.

File: src/pages/homepage/components/main-feed/MainFeed.css [hunk 10]
```diff
@@ -0,0 +1,14 @@
+.main-feed{
+    /* display: grid;
+    grid-template-columns: 1fr;
+    grid-template-areas: 
+    "moments"
+    "create-post"
+    "posts"; */
```
Comment (zoef-yt) [inline src/pages/homepage/components/main-feed/MainFeed.css:7]: remove commented code.

File: src/pages/homepage/components/post/Post.tsx [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    avatar: "MB",
+    name: "Muskan Bagrecha",
+    comment: "Nice work musk!",
+    date: "2020-01-01",
+  });
+
+  return (
+    <article className="post rounded-md bg-primary-200 m-2 py-2">
+      <header>
+        <div>
```
Comment (zoef-yt) [inline src/pages/homepage/components/post/Post.tsx:17]: This extra div can be removed.

File: src/pages/homepage/components/post/Post.tsx [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+          <p className="text-base">Muskan Bagrecha</p>
+          <p className="text-xs">12 hours ago</p>
+        </div>
+      </header>
+      <img
+        src="https://images.unsplash.com/photo-1556227691-add1cf868ff6?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1441&q=80"
+        alt=""
+      />
+      <footer className="p-2">
+        <div className="flex justify-between"></div>
```
Comment (zoef-yt) [inline src/pages/homepage/components/post/Post.tsx:30]: This extra div can be removed.

**Review by zoef-yt — CHANGES_REQUESTED** [review]

File: src/pages/homepage/components/moments/Moments.css [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+@media screen and (max-width: 640px) {
+    .moments{
+        height: 6rem;
+        padding: 5px;
+    }
+    
+    .moments > img{
+        height: 5rem;
+        width: 5rem;
+        clip-path: circle();
```
Comment (rahulyadav139) [inline src/pages/homepage/components/moments/Moments.css:27]: Appreciation: great you have used the circle method.

File: src/pages/homepage/components/post/Post.tsx [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+          <div className="avatar avatar-xs avatar-text rounded">MB</div>
+        </div>
+        <div className="flex flex-col">
+          <p className="text-base">Muskan Bagrecha</p>
+          <p className="text-xs">12 hours ago</p>
+        </div>
+      </header>
+      <img
+        src="https://images.unsplash.com/photo-1556227691-add1cf868ff6?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1441&q=80"
+        alt=""
```
Comment (rahulyadav139) [inline src/pages/homepage/components/post/Post.tsx:27]: alt tag is empty. Please give some value to this tag.

**Review by rahulyadav139 — APPROVED** [review]

**Comment by muskanbagrecha** [comment]
All comments will be addressed in subsequent PRs.

**Pull request merged**